flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -1621,6 +1621,11 @@ TaskExecutorToResourceManagerConnection getResourceManagerConnection() {
 		return resourceManagerHeartbeatManager;
 	}
 
+	@VisibleForTesting
+	SlotReport getSlotReport() {
```

Comment:
You can probably drop this method if you move the code that computes the slot resources into a separate method:
`computeSlotResourceProfile(taskManagerServicesConfiguration)` as suggested. Then you can test that logic in isolation and do not need access to the slot requests.